### PR TITLE
Streaming API error handling/reporting improvements

### PR DESCRIPTION
**Pull request by leo-da**
Items 1 and 2 of: #4521 

question about error reporting:
https://github.com/akka/akka-http/issues/3023

### TODO
- [x] Update docs
- [ ] Add test for binary message input

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/json-api/index.rst [hunk 1]
```diff
@@ -1053,6 +1055,54 @@ Streaming API
...
+    const tokenPrefix = "jwt.token."
+    const wsProtocol = "daml.ws.auth"
+    const jwt =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJmb29iYXIiLCJhY3RBcyI6WyJBbGljZSJdfX0.VdDI96mw5hrfM5ZNxLyetSVwcD7XtLT4dIdHIOa9lcU"
+    const subprotocol = `${tokenPrefix}${jwt},${wsProtocol}`
+
+    const ws = new WebSocket("ws://localhost:7575/v1/stream/query", [subprotocol])
+
+    ws.on("open", function open() {
+      ws.send(`{"templateIds": ["Iou:Iou"]}`)
```
Comment (S11001001) [inline docs/source/json-api/index.rst]: I think you would typically write the JS object as an object and then encode it to JSON, yes?

Comment (leo-da) [reply docs/source/json-api/index.rst]: this is Javascript, no types who cares :) but yeah `JSON.stringify` should do the trick. I will change it.

Comment (leo-da) [reply docs/source/json-api/index.rst]:
```
    ws.on("open", function open() {
      ws.send(JSON.stringify({templateIds: ["Iou:Iou"]}))
    })
```

**Comment by leo-da** [comment]
@hurryabit I am going to merge it as is. The doc update will be in a separate PR.

File: docs/source/json-api/index.rst (hunk 1 above)
Comment (S11001001) [reply docs/source/json-api/index.rst]: For sure :) you can at least catch dumb syntax typos like missing `}` or `"` this way.

**Pull request merged**